langchain4j/src/main/java/dev/langchain4j/store/embedding/chroma/ChromaEmbeddingStore.java

```diff
@@ -0,0 +1,172 @@
+                        ? null
+                        : textSegments.stream()
+                        .map(TextSegment::metadata)
+                        .map(Metadata::asMap)
+                        .collect(toList()))
+                .documents(textSegments == null
+                        ? null
+                        : textSegments.stream()
+                        .map(TextSegment::text)
+                        .collect(toList()))
+                .build();
+
+        chromaClient.addEmbedding(collectionId, embeddingsRequest);
+    }
+
+    @Override
+    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults) {
+        return findRelevant(referenceEmbedding, maxResults, 0);
+    }
+
+    @Override
+    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults, double minScore) {
+        float[] vector = referenceEmbedding.vector();
+        QueryRequest queryRequest = QueryRequest.builder()
+                .queryEmbedding(singletonList(IntStream.range(0, vector.length)
+                        .mapToObj(i -> vector[i])
+                        .collect(toList())))
+                .nResults(maxResults)
+                .build();
+
+        SuccessfulResponse nearestNeighbors = chromaClient.getNearestNeighbors(collectionId, queryRequest);
+
+        return toEmbeddingMatches(nearestNeighbors);
+    }
+
+    private static List<EmbeddingMatch<TextSegment>> toEmbeddingMatches(SuccessfulResponse nearestNeighbors) {
+        List<EmbeddingMatch<TextSegment>> embeddingMatches = new ArrayList<>();
+        for (int i = 0; i < nearestNeighbors.getIds().get(0).size(); i++) {
+            EmbeddingMatch<TextSegment> textSegmentEmbeddingMatch = new EmbeddingMatch<>(
+                    distanceToScore(nearestNeighbors.getDistances().get(0).get(i)),
+                    nearestNeighbors.getIds().get(0).get(i),
+                    Embedding.from(nearestNeighbors.getEmbeddings().get(0).get(i)),
+                    nearestNeighbors.getDocuments().get(0).get(i) == null ? null : TextSegment.from(nearestNeighbors.getDocuments().get(0).get(i),
+                            nearestNeighbors.getMetadatas().get(0).get(i) == null ? null : new Metadata(nearestNeighbors.getMetadatas().get(0).get(i)))
+            );
+            embeddingMatches.add(textSegmentEmbeddingMatch);
+        }
+        return embeddingMatches;
+    }
+
+    /**
+     * By default, cosine distance will be used. For details: <a href="https://github.com/nmslib/hnswlib/tree/master#python-bindings"></a>
+     * Converts a cosine distance in the range [0, 2] to a score in the range [1, 0].
+     *
+     * @param distance The distance value.
+     * @return The converted score.
+     */
+    private static double distanceToScore(double distance) {
+        if (distance < 0 || distance > 2) {
+            throw new IllegalArgumentException("Distance must be in the range [0, 2]");
```

Comment:
after you hardcoded distance metric during collection creation, does it return something out of range [0,2]?

Reply:
done